ibis/backends/base/sql/__init__.py

```diff
@@ -204,7 +204,7 @@ def to_pyarrow_batches(
 
         return pa.ipc.RecordBatchReader.from_batches(schema.to_pyarrow(), batches)
 
-    def _compile_udfs(self, expr: ir.Expr) -> Iterable[str]:
+    def _compile_udfs(self, expr: ir.Expr) -> Iterable[str | Callable[..., None]]:
         """Return an iterator of DDL strings, once for each UDFs contained within `expr`."""
```

Comment:
If you don't do the refactor I suggested below, we'd at least want to update this docstring to be correct.

Reply:
Did the refactor!